pkg/controller/installation/products/fuse/reconciler.go

```diff
@@ -0,0 +1,212 @@
+	// NEED to ensure a subscription is created if not exists
+	// need to make sure there is only one operator source
+	r.logger.Infof("reconciling subscription %s from channel %s in namespace: %s", defaultSubscriptionName, "integreatly", r.Config.GetNamespace())
+	err := r.mpm.CreateSubscription(
+		marketplace.GetOperatorSources().Integreatly,
+		r.Config.GetNamespace(),
+		defaultSubscriptionName,
+		marketplace.IntegreatlyChannel,
+		[]string{r.Config.GetNamespace()},
+		v1alpha12.ApprovalAutomatic)
+	if err != nil && !errors2.IsAlreadyExists(err) {
+		return v1alpha1.PhaseFailed, errors.Wrap(err, fmt.Sprintf("could not create subscription in namespace: %s", r.Config.GetNamespace()))
+	}
+	return r.handleAwaitingOperator(ctx)
+}
+
+func (r *Reconciler)reconcileCustomResource(ctx context.Context, install *v1alpha1.Installation)(v1alpha1.StatusPhase, error)  {
+	ref := v12.NewControllerRef(install,v1alpha1.SchemaGroupVersionKind)
+	intLimit := -1
+	cr := syn.Syndesis{
+		ObjectMeta: v12.ObjectMeta{
+			OwnerReferences:[]v12.OwnerReference{
+				*ref,
+			},
+			Namespace:r.Config.GetNamespace(),
+			Name:"integreatly",
+		},
+		TypeMeta: v12.TypeMeta{},
+		Spec:syn.SyndesisSpec{
+			Integration:syn.IntegrationSpec{
+				Limit: &intLimit,
+			},
+			Components:syn.ComponentsSpec{
+				Server:syn.ServerConfiguration{
+					Features:syn.ServerFeatures{
+						ExposeVia3Scale:true,
+					},
+				},
+			},
+		},
+	}
+	if err := r.client.Get(ctx,pkgclient.ObjectKey{Name: cr.Name},&cr); err != nil{
+		if errors2.IsNotFound(err){
+			if err := r.client.Create(ctx,&cr); err != nil && !errors2.IsAlreadyExists(err){
+				return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to create a syndesis cr when reconciling cutom resource")
+			}
+			return v1alpha1.PhaseNone,nil
+		}
+		return v1alpha1.PhaseFailed, errors.Wrap(err, "failed to get a syndesis cr when reconciling cutom resource")
+	}
+	if cr.Status.Phase != syn.SyndesisPhaseInstalled{
+		return v1alpha1.PhaseInProgress, nil
+	}
+	if cr.Status.Phase == syn.SyndesisPhaseStartupFailed{
+		return v1alpha1.PhaseFailed, errors.New("syndesis has failed to install " + string(cr.Status.Reason))
+	}
+	return v1alpha1.PhaseNone, nil
+}
+
+func (r *Reconciler) handleAwaitingOperator(ctx context.Context) (v1alpha1.StatusPhase, error) {
+	r.logger.Infof("checking installplan is created for subscription %s in namespace: %s", defaultSubscriptionName, r.Config.GetNamespace())
+	ip,sub, err := r.mpm.GetSubscriptionInstallPlan(defaultSubscriptionName, r.Config.GetNamespace())
+	if err != nil {
+		logrus.Infof("error in handleAwaitingOperator ", err.Error())
+		if errors2.IsNotFound(err) {
+			logrus.Infof("error in handleAwaitingOperator is not found error ")
+			if sub != nil{
+				logrus.Infof("time since created %v", time.Now().Sub(sub.CreationTimestamp.Time).Seconds())
+			}
+			if sub != nil && time.Now().Sub(sub.CreationTimestamp.Time) > time.Second * 60{
```

Comment:
it gives up waiting for the installplan after a minute? I've seen it work after a longer pause than that, I'd be more inclined to put this at around 5 minutes, and in the fuse config.

Reply:
This isnt specific to fuse.

Reply:
moved to config package